runtime/buffer.cc

```diff
@@ -64,7 +66,90 @@ raw_buffer_ptr raw_buffer::make_copy(const raw_buffer& src) {
   return buf;
 }
 
+namespace {
+
+// This algorithm is the same idea as [1], but here it does not need to attempt to support compile-time constant
+// strides, which makes it a lot easier to implement.
+//
+// 1. https://github.com/dsharlet/array/blob/38f8ce332fc4e26af08325ad0654c8516a445e8c/include/array/array.h#L835-L907
+
+// A proposed stride is "OK" w.r.t. `dim` if the proposed stride does not cause this dimension's memory to overlap with
+// any other dimension's memory.
+bool is_stride_ok(index_t stride, index_t extent, const dim& dim) {
+  if (dim.stride() == dim::auto_stride) {
+    // If the dimension has an unknown stride, it's OK, we're
+    // resolving the current dim first.
+    return true;
+  } else if (extent == 1 && abs(stride) == abs(dim.stride()) && alloc_extent(dim) > 1) {
```

Comment:
We should either fix or disable these warnings

Reply:
Definitely legit warnings, I wish I got such warnings when building locally!